src/player.cpp

```diff
@@ -6965,6 +6988,15 @@ bool player::has_charges( const itype_id &it, int quantity,
     if( it == "fire" || it == "apparatus" ) {
         return has_fire( quantity );
     }
+    if( it == "UPS" && is_mounted() &&
+        mounted_creature.get()->has_flag( MF_RIDEABLE_MECH ) ) {
+        auto mons = mounted_creature.get();
+        if( quantity <= mons->battery_item->ammo_remaining() ) {
+            return true;
+        } else {
+            quantity -= mons->battery_item->ammo_remaining();
+        }
```

Comment:
And here, you also continue to count the other sources of UPS charges, and will return true when there are enough there.